File: Data Structures & Algorithms/duplicate-integer/submission-7.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
     Set uniqueNums = new HashSet<Integer>();
    
     for (int i: nums) {
        uniqueNums.add(i);
     }
    
    return (uniqueNums.size() != nums.length);
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-7.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l: int = 0
        r: int = len(nums) - 1

        while l < r:
            mid: int = (l + r) // 2
            print(f'l: {l}, r: {r}, mid: {mid}')

            if nums[mid] > nums[r]:
                l = mid + 1
            else:
                r = mid

        return nums[l]